polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -316,19 +324,25 @@ async fn handle_parent_process(
 	mut pipe_read: os_pipe::PipeReader,
 	temp_artifact_dest: PathBuf,
 	worker_pid: u32,
+	usage_before: Usage,
+	timeout: u64,
 ) -> Result<PrepareStats, PrepareError> {
 	let mut received_data = Vec::new();
 
-	pipe_read.read_to_end(&mut received_data).map_err(|_| {
-		PrepareError::Panic(format!("error reading pipe for worker id {}", worker_pid))
-	})?;
-	return match nix::sys::wait::wait() {
+	pipe_read
+		.read_to_end(&mut received_data)
+		.map_err(|err| PrepareError::Panic(err.to_string()))?;
+	let status = nix::sys::wait::wait();
+	let usage_after = nix::sys::resource::getrusage(UsageWho::RUSAGE_CHILDREN)
```

Comment:
Awesome! I would just add a comment here, saying that this is needed to determine if `setrlimit` was triggered.